doc/pins/PINS_SONiC_Design_for_SaiGenericExt.md

```diff
@@ -0,0 +1,429 @@
+# PINS (P4 Integrated Network Stack) SONiC Control Plane for SAI Generic Extensions
+
+## Table of Content
+- [Introduction](#Itroduction)
+- [Definitions](#Definitions)
+- [Scope](#Scope)
+- [Requirements](#Requirements)
+- [Highlevel Design](#Highlevel-Design)
+  - [P4RT-APP](#P4RT-APP)
+  - [P4Orch](#P4Orch)
+    - [Table Definition database](#Table-Definition-database)
+    - [Demultiplexing SAI Extension entries](#Demultiplexing-SAI-Extension-entries)
+    - [SAI Extension Manager](#SAI-Extension-Manager)
+    - [Table processing precedence](#Table-processing-precedence)
+    - [Cross Table dependencies](#Cross-Table-dependencies)
+  - [Application Level Response](#Application-Level-Response)
+  - [Counters and Meters](#Counters-and-Meters)
+  - [CRM (Critical Resource Monitoring)](#CRM)
+  - [Warm restart](#Warm-restart)
+- [Appendix](#Appendix)
+  - [P4 Program for SAI Extensions and useful annotations](#P4-Program-annotations)
+  - [Example](#Example)
+
+
+## Revision
+| Rev  |   Date    |                        Author                        | Change Description    |
+| :--: | :-------: | :--------------------------------------------------: | :--------------------:|
+| 0.1  | 9/27/2022 | Intel - Shitanshu Shah, Reshma Sudarshan,            |  Initial Revision     |
+|      |           |         Brian O’Connor, Ravi Vantipalli, Sagar Balani|                       |
+|      |           | Google - Bhagat Janarthanan, Kishan, Runming Wu,     |                       |
+|      |           |          Robert Halstead, Kishore Gummadidala        |                       |
+| 0.2  | 9/29/2022 | Shitanshu Shah                                       |address review comments|
+
+## Introduction <a name="Introduction"></a>
+P4 is a language for expressing how packets are processed by the data plane of a programmable network forwarding device. P4 programs specify how the various programmable blocks of a target architecture are programmed and connected. PINS (P4 Integrated Network Stack) is an architecture, described in the [PINS HLD](https://github.com/sonic-net/SONiC/blob/master/doc/pins/pins_hld.md), to implement a P4 program, in the SONiC framework, on top of the SAI pipeline. SAI pipeline is a well-documented forwarding behavior of a data-plane. In PINS architecture, a \set of standard SAI capabilities are modeled in P4 and configurable behaviors, like ACLs, are defined using P4.

```

Comment:
a \set ==> a set

Reply:
Hi Lihua, thank you very much for careful reviewing. We will certainly incorporate this change you have suggested.